doc/paper/paper.tex

```diff
@@ -0,0 +1,1012 @@
+    interface it is possible to list tables, view statistics, and run interactive
+    queries.
+
+    Communication between client and server is over an HTTP API, with custom HTTP
+    headers in use to transmit the versions of streamed tables down to the client
+    during query operations. We view this API as fairly ad-hoc and temporary, and
+    intend to either firm it up or switch to grpc once we are in a better position
+    to evaluate.
+
+    \section{Discussion}
+
+    \subsection{Related Work}
+    The copy-on-write tree structure of dp3 borrows heavily from btrdb, a
+    timeseries database for storing streams of high-frequency univariate floats.
+    Comparing the two, btrdb is more of a \q{pure} timeseries database with a
+    primary focus on analytics, while dp3 has a primary focus on row-oriented
+    multivariate data playback and a secondary focus on analytics. Since btrdb is
+    univariate, it is also effectively a column store and is able to make use of
+    columnar compression techniques that are not available to dp3, which uses
+    chunked zstd compression across rows.
+
+    Another system that uses a similar storage scheme is Apache Iceberg
+    \cite{iceberg}.  Iceberg is a table format specification rather than an
+    implementation, but the format specifies a height-three copy-on-write tree
+    consisting of metadata files, manifest lists, manifest files, and data files.
+    The metadata is a JSON file storing a list of \q{snapshots} that reference
+    manifest lists.  Manifest lists include summary information about manifest
+    files, and manifest files contain a list of data files in Avro format.
+    Iceberg has built-in support for Parquet, Avro, and ORC format data files. As
+    with dp3 and btrdb, it seems logical in deployment scenarios to cache the
+    upper nodes of the tree.
+
+    The primary intent of Iceberg is to enable multiple database engines to
+    operate against the same data safely at the same time, but as a useful
+    side-effect of having data in Iceberg format you gain the ability to use an
+    off-the-shelf database engine for free, and the potential for pluggable
+    integrations in the future if more databases add support for external Iceberg
+    tables, as Snowflake has recently done. In addition, dp3 may at some point
+    want to have multiple engines operating at the same time, for instance if we
+    were able to integrate with Spark SQL. These possibilities make Iceberg a
+    worthwhile direction to explore as backing storage for dp3. In comparing the
+    two, Iceberg stands out for its far more developed SQL and transactional
+    capabilities, while dp3 is distinguished in supporting message storage and
+    playback in the user's native choice of recording format, without on-the-fly
+    transcoding during read. dp3 also has a more granular time index due to
+    having a deeper tree.
```

Comment:
Wondering why Iceberg takes up more space here than btrdb. Is this intended? Hasn't the project been more closely aligned with btrdb?